### ISSUE TITLE
Can't test finaid app on dev server

### ISSUE THREAD

**Issue by mgersh** [learning-unlimited/ESP-Website#1734]
secure_required forces it to https, and dev servers can't handle https. I think this is now redundant with other parts of our production setup and we can remove it.

**Comment by benjaminjkraft** [comment]
We could also just bypass it if DEBUG is true.

**Comment by willgearty** [comment]
Should be fairly easy to add a bypass here:
https://github.com/learning-unlimited/ESP-Website/blob/d3240e08fdd825420e5a9a896b539e67a9a238ef/esp/esp/utils/web.py#L134

**Comment by Oval17** [comment]
Hi @willgearty ,
Would like to work on this.
Thanks!!

**Comment by willgearty** [comment]
@Oval17 please work on your assigned issues before getting assigned to more.

**Comment by Oval17** [comment]
Hi @willgearty , 
Yes sure just a heads up have got to know about this organisation from GSoC page, also just to be sure GSoC is open for working professionals as well, 
It might be confusing but I am not new to development have experience of couple of years. 
Though apologies for working without assignment but nowhere it was mentioned neither on contributers page of GSoC so I thought we can pick and work.
Will keep in mind and am really excited to work with this org ☺️
Thanks

**Comment by willgearty** [comment]
We're normally pretty relaxed here about working on whatever you want, but things recently got a bit busier with our acceptance into GSoC, so I'm trying to enforce a bit more formality (and especially want to make sure there are plenty of starter issues for all of the new contributors).

From the [GSoC website](https://summerofcode.withgoogle.com/rules):

> To participate in the Program, a GSoC Contributor must:
> - be eighteen (18) years of age or older upon registration for the Program;
> - for the duration of the Program, be eligible to work in the country in which they reside;
> - not be an Organization Administrator or Mentor in the Program; and
> - be a student or a beginner to open source software development.

I'm not a GSoC program organizer, but I imagine working professionals with years of development and open source experience are probably not eligible. I understand you're probably here for the GSoC $$, so I understand if you're no longer interested, but if not, we're happy to have you! I've opened up the [discussions page](https://github.com/learning-unlimited/ESP-Website/discussions) if you have questions that aren't tied to a particular issue.

**Comment by Oval17** [comment]
Yes sure, it says student , I am working professional but I am still in college completing my undergrad degree.
And yes I am interested and would contribute fairly to the org according to our plan.
Will love to apply for GSoC as well, I guess would love to share plan on bootstrap bump.
@willgearty would like to implement things faster as well so that we can fast track our process given the advent of AI in coding all we need is rigorous review can set up Copilot reviewer as well

Thanks

**Comment by willgearty** [comment]
Appreciate the enthusiasm! I maintain this project in my off-time, so I can't guarantee very quick turnaround on some discussions, but will try my best.

We currently require that at least one core [human] contributor reviews each pull request. We may use Copilot to speed this up, but there will always be a human in the loop to ensure codebase additions aren't malicious and work properly. Since our infrastructure is used by thousands of users across the US, we need to ensure that changes work smoothly before pushing them to production. This means spinning them up in local development servers and performing manual testing.

**Comment by Oval17** [comment]
Thanks
I have worked on several small bugs as you mentioned more can be assigned..
Just to say my plan is I see a lot of issues in our repo though fair analysis would be if everyone works on them equally but I will keep on working on them as assigned to me.
Also Just give me few days I will go through more how infra structurally code is written and as I get deeper understanding would be more easier.
Also will share my plan for bootstrap bump soon I guess its something important to us.
